File: AtlasWeb/Core/Modules/Core_Destek_Modulleri/Test_Modulu/ZForm1ViewModel.cs
//$BF3682A0
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class ZDeneme1ServiceController
    {
        partial void PostScript_ZForm1(ZForm1ViewModel viewModel, ZDeneme1 zDeneme1, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            if (viewModel._ZDeneme1.Name.Substring(0, 1) == "A" && transDef.ObjectDefID == ZDeneme1.States.Approve)
                throw new Exception();
        }
    }
}

namespace Core.Models
{
    public partial class ZForm1ViewModel: BaseViewModel
    {
       

    }
}

File: AtlasWeb/Core/Modules/Core_Destek_Modulleri/Test_Modulu/ZForm1ViewModel.ts
//$4F18D3AF
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { ZDeneme1 } from "NebulaClient/Model/AtlasClientModel";
import { City } from "NebulaClient/Model/AtlasClientModel";

export class ZForm1ViewModel extends BaseViewModel {
    public _ZDeneme1: ZDeneme1 = new ZDeneme1();
    public Citys: Array<City> = new Array<City>();
}
